lib/segment/src/spaces/simple.rs

```diff
@@ -28,21 +28,22 @@ impl Metric for EuclidMetric {
     fn similarity(&self, v1: &[VectorElementType], v2: &[VectorElementType]) -> ScoreType {
         #[cfg(target_arch = "x86_64")]
         {
-            if is_x86_feature_detected!("avx") && is_x86_feature_detected!("fma") {
+            if is_x86_feature_detected!("avx") && is_x86_feature_detected!("fma") && v1.len() >= 32
+            {
                 return unsafe { euclid_similarity_avx(v1, v2) };
             }
         }
 
         #[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
         {
-            if is_x86_feature_detected!("sse") {
+            if is_x86_feature_detected!("sse") && v1.len() >= 16 {
                 return unsafe { euclid_similarity_sse(v1, v2) };
             }
         }
 
         #[cfg(all(target_arch = "aarch64", target_feature = "neon"))]
         {
-            if std::arch::is_aarch64_feature_detected!("neon") {
+            if std::arch::is_aarch64_feature_detected!("neon") && v1.len() >= 16 {
```

Comment:
Maybe move `16` into a const: `MIN_V_SIZE_SIMD`?

Reply:
It would be also great to have `When the dimension is lower than simd processing stride, it is faster to use regular functions instead of simd` this as a doc comment on top of the const

Reply:
Moved `32` and `16` magic numbers into consts. Don't sure about such comment